libafl_libfuzzer/libafl_libfuzzer_runtime/src/corpus.rs

```diff
@@ -0,0 +1,321 @@
+use std::{
+    cell::RefCell,
+    collections::{hash_map::Entry, BTreeMap, HashMap},
+    io::ErrorKind,
+    path::PathBuf,
+    sync::atomic::{AtomicU64, Ordering},
+};
+
+use libafl::{
+    corpus::{inmemory::TestcaseStorage, Corpus, CorpusId, Testcase},
+    inputs::{Input, UsesInput},
+};
+use libafl_bolts::Error;
+use serde::{Deserialize, Serialize};
+
+/// A corpus which attempts to mimic the behaviour of libFuzzer.
```

Comment:
So what's different about a LibFuzzer corpus

Reply:
- no lock files
- reject replace/remove
- preservation of existing files

There's also the `ArtifactCorpus` which just doesn't keep track at all except for the very last solution.

Reply:
Also, this seems to be significantly faster -- local quick test shows the average performance with large corpora to be about 2.5x CachedOnDisk.